src/google/protobuf/compiler/cpp/cpp_message.cc

```diff
@@ -512,15 +510,11 @@ class ColdChunkSkipper {
   const std::vector<std::vector<const FieldDescriptor*>>& chunks_;
   const std::vector<int>& has_bit_indices_;
   const AccessInfoMap* access_info_map_;
-  const double cold_threshold_;
```

Comment:
@keith @acozzette two possible options:

Add `__attribute__((used))` to this field.

Or, wrap this in:

```c++
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wunused-whatever
const double cold_threshold_;
#pragma clang diagnostic pop
```

Reply:
> Add __attribute__((used)) to this field. I haven't tried this, not sure if it works in all places, specifically member fields.

Turns out it does not

> #pragma clang diagnostic push

Are these ok when this is being compiled with gcc?

Reply:
`#pragma GCC diagnostic ...` would also have to be added to support gcc.